x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/hosts/all/index.ts

```diff
@@ -62,7 +62,7 @@ export const allHosts: SecuritySolutionFactory<HostsQueries.hosts> = {
     };
     const showMorePagesIndicator = totalCount > fakeTotalCount;
 
-    const hostNames = buckets.map(getOr('', 'key'));
+    const hostNames = edges.map((edge) => getOr('', 'node.host.name[0]', edge));
```

Comment:
Use edges instead of buckets because buckets contain all hosts, not only those on the current page.